ce/ce/archivers/ZipUnpacker.ts

```diff
@@ -49,6 +49,19 @@ export class ZipUnpacker extends Unpacker {
 
       const openedFile = await archiveUri.openFile();
       const zipFile = await ZipFile.read(openedFile);
+      if (options.strip === -1) {
+        // when strip == -1, strip off all common folders off the front of the file names
+        options.strip = 0;
+        const folders = [...zipFile.folders.keys()].sort((a, b) => a.length - b.length);
+        const files = [...zipFile.files.keys()];
+        for (const folder of folders) {
+          if (files.all((filename) => filename.startsWith(folder))) {
+            options.strip = folder.split('/').length - 1;
```

Comment:
Do we need to worry about \\?

Reply:
I don't believe so - all zip files folder names I've found had forward slashes - but I can test and see if there is a chance that back slashes are in them.

Reply:
Forward slashes are required for paths in zip files in the spec. 

![image](https://user-images.githubusercontent.com/133736/175719091-708136fb-0591-4c51-a099-0fc8d94e9312.png)